### PR TITLE
Change scatter plot in patient view to D3

### PR DESCRIPTION
**Pull request by yichaoS**
http://dashi-dev.cbio.mskcc.org:8080/patient-view-d3-plots/

### CODE REVIEW

File: portal/src/main/webapp/WEB-INF/jsp/tumormap/cancer_study_view/mut_cna_scatter_plot.jsp [hunk 1]
```diff
@@ -32,28 +32,28 @@
...
-                    V-Axis scale: <input type="radio" name="mut-cna-vaxis-log" class="mut-cna-axis-log" value="normal" checked="checked"/>linear &nbsp;
-                    <input type="radio" name="mut-cna-vaxis-log" class="mut-cna-axis-log" value="log" id="mut-cna-vaxis-log"/>log
-                </td>
-                <td id="case-id-div" align="right">
-                </td>
-        </tr>
-    </table>
-</fieldset>
\ No newline at end of file
+    <%--<table style="display:block;width:100%;" class="mut-cna-config">--%>
```
Comment (yichaoS) [inline portal/src/main/webapp/WEB-INF/jsp/tumormap/cancer_study_view/mut_cna_scatter_plot.jsp]: Commented lines could be removed here.

File: portal/src/main/webapp/js/src/co-exp/components/ScatterPlots.js [hunk 2]
```diff
@@ -242,35 +305,95 @@ var ScatterPlots = function() {
...
+            elem.axisTitleGroup.append("svg:foreignObject")
+                .attr("id", "plots-title-y-checkbox")
+                .attr("transform", "rotate(-90)")
+                .attr("x", (canvas.yTop - canvas.yBottom) / 2 + _yTitle.length / 2 * 8-16)
+                .attr("y", canvas.xLeft - 75)
+                .attr("width", "50")
+                .attr("height", "20")
+                .append("xhtml:body")
+                .style({"font-size": "11px", "margin": "0"})
+                .html("<input id='mut-cna-vaxis-log' class='mut-cna-axis-log' type='checkbox' style='float:left' "+_checked+"/><span style='float:left; margin-top: 2px; font-size-adjust: 0.5;'>Log</span>");
```
Comment (yichaoS) [inline portal/src/main/webapp/js/src/co-exp/components/ScatterPlots.js]: Kind of redundant to define variable "_checked" here

**Comment by yichaoS** [comment]
- Style inconsistent when zoom in
- Patient view link in the patient view, is linking to using old plots

**Comment by jjgao** [comment]
@JiaoJiao123 @yichaoS what's the status of this PR? I am thinking of merge it soon.

**Comment by yichaoS** [comment]
@jjgao @JiaoJiao123 I did the code review. And we deployed a test war file. http://dashi-dev.cbio.mskcc.org:8080/patient-view-d3-plots/  Could you please help us to do the feature review?

**Comment by jjgao** [comment]
@yichaoS It looks like that it is still based on Google Charts in the dashi-dev war. Could you double check?

**Comment by yichaoS** [comment]
@jjgao I just updated the test war file again. It should work now. Sorry about that. And also, I noticed once zoom in, the style of the plots are different (color, stroke, etc). Do you think we need to keep the consistency?

**Comment by jjgao** [comment]
@yichaoS @JiaoJiao123: it looks good to me. Let's release this for now. We should refactor this again with plotly at certain point.

**Pull request merged**